tasks/nginx.yml

```diff
@@ -18,10 +18,19 @@
     src: ../templates/nginx.j2
     dest: "/etc/nginx/sites-available/{{ inventory_hostname }}"
   notify: restart nginx
+  when: ansible_distribution != 'CentOS' and ansible_distribution != 'Red Hat Enterprise Linux'
 
 - name: nginx site enabled
   file:
     src: "/etc/nginx/sites-available/{{ inventory_hostname }}"
     dest: "/etc/nginx/sites-enabled/{{ inventory_hostname }}"
     state: link
   notify: restart nginx
+  when: ansible_distribution != 'CentOS' and ansible_distribution != 'Red Hat Enterprise Linux'
+
+- name: nginx site
+  template:
+    src: ../templates/nginx.j2
+    dest: "/etc/nginx/conf.d/{{ inventory_hostname }}.conf"
```

Comment:
Unfortunately in case of Debian it's a link to file in `sites-enabled`. In RedHats this is plain file. So variables won't eliminate task for RedHat.